### DISCUSSION TITLE
Partition with condition

### DISCUSSION THREAD

**Discussion by guanghuang** [pgpartman/pg_partman#454] [Ideas]
Try to partition by customer id, but only some customers have lots of data (e.g > 1m), don't want to create partition for all customers, only for the customers which have large data. Is it possible that when run_maintenance only create child partition when the customer has count of data larger than 1m?

**Comment by keithf4** [comment]
That is not something currently supported in pg_partman. This extension only works with fixed interval values at this time. It's likely technically possible in PostgreSQL, but would be very challenging to support something like this in a general way that pg_partman could support.
